test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -257,6 +257,28 @@ public static RequestOptions expectVersionSpecificWarnings(Consumer<VersionSensi
     public static RequestOptions expectWarnings(String... warnings) {
         return expectVersionSpecificWarnings(consumer -> consumer.current(warnings));
     }
+    
+    /**
+     * Creates RequestOptions designed to ignore [types removal] warnings but nothing else 
+     * @deprecated this method is only required while we deprecate types and can be removed in 8.0
+     */
+    @Deprecated
+    public static RequestOptions expectTypeRemovalWarnings() {
```

Comment:
Totally up to you, but I'm wondering if the name `allowTypesRemovalWarnings` would make more sense?